FileSystemSimulator/Shell/Shell.cpp

```diff
@@ -0,0 +1,251 @@
+#include "pch.h"
+#include "Shell.h"
+#include <iostream>
+
+using namespace std;
+
+Shell::Shell(FileSystem* i_filesystem)
+{
+	filesystem = i_filesystem;
+}
+
+Shell::Shell()
+{
+	filesystem = new FileSystem();
+}
+
+Shell::~Shell()
+{
+	delete filesystem;
+}
+
+void Shell::printHelp()
+{
+	cout << "Help:\nCommands:\n";
+	cout << "create file - " + create_command + " <filename>\n";
+	cout << "destroy file - " + destroy_command + " <filename>\n";
+	cout << "open file - " + open_command + " <filename>\nnote: open file command " + 
+		"returns opened file key, use it to read, write and lseek through the file\n";
+	cout << "close file - " + close_command + " <key>\n";
+	cout << "read from file - " + read_command + " <key> <number_of_characters_to_read>\n";
+	cout << "write to file - " + write_command + " <key> <text> <number_of_characters_to_write>\n";
+	cout << "lseek in file - " + lseek_command + " <key> <position>\n";
+	cout << "list all files on disk - " + directory_command + "\n";
+	cout << "exit simulator - " + exit_command + "\n\n";
+}
+
+int Shell::parseCommand(string i_command_string)
+{
+	if (!isValidCommandName(i_command_string.substr(0, 3))) {
+		cout << "Invalid command name, try again. Print gh to get help and view list of commands.\n";
+		return invalid_command_code; 
+	}
+
+	if (i_command_string.substr(0, 2) == exit_command) {
+
+		cout << "End.\n";
+		return exit_code;
+	}
+
+	if (i_command_string.substr(0, 2) == help_command) {
+
+		printHelp();
+		return success_code;
+	}
+
+	if (i_command_string.substr(0, 2) == create_command) {
+
+		i_command_string.erase(0, 3);
+		printCreateCommandResult(i_command_string);
+	}
+
+	if (i_command_string.substr(0, 2) == destroy_command) {
+
+		i_command_string.erase(0, 3);
+		printDestroyCommandResult(i_command_string);
+	}
+
+	if (i_command_string.substr(0, 2) == open_command) {
+
+		i_command_string.erase(0, 3);
+		printOpenCommandResult(i_command_string);
+	}
+
+	if (i_command_string.substr(0, 2) == close_command) {
+
+		i_command_string.erase(0, 3);
+		int key = getKeyFromCommandString(i_command_string);
+		printCloseCommandResult(key);
+	}
+
+	if (i_command_string.substr(0, 2) == read_command) {
+
+		i_command_string.erase(0, 3);
+		int key = stoi(getIWord(i_command_string, 1));
+		int number_of_chars = stoi(getIWord(i_command_string, 2));
+		printReadCommandResult(key, number_of_chars);
+	}
+
+	if (i_command_string.substr(0, 2) == write_command) {
+
+		i_command_string.erase(0, 3);
+		int key = getKeyFromCommandString(i_command_string);
+		string text = getIWord(i_command_string, 2);
+		int number_of_characters = stoi(getIWord(i_command_string, 3));
```

Comment:
we do not need number of characters, the length is uniquely determined by string (text)